rust/lance-arrow/src/deepcopy.rs

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright The Lance Authors
+
+use std::sync::Arc;
+
+use arrow_array::{make_array, Array, RecordBatch};
+use arrow_buffer::{Buffer, NullBuffer};
+use arrow_data::ArrayData;
+
+pub fn deep_copy_buffer(buffer: &Buffer) -> Buffer {
+    Buffer::from(Vec::from(buffer.as_slice()))
```

Comment:
Do we need to copy data here?
Can we just do `buffer.data.clone()`?

https://docs.rs/arrow-buffer/51.0.0/src/arrow_buffer/buffer/immutable.rs.html#34

Reply:
We cannot. When the buffer is imported over FFI, the refcount of the buffer is tied to the entire batch. Thus, it can't be freed until all other columns in the same batch are also ready to be freed. By copying the buffer into Rust, we gain control over the lifetime and can free it earlier.

Like Weston said in the PR description, we can solve this by importing each column individually over FFI.